src/sparseml/benchmark/info.py

```diff
@@ -0,0 +1,520 @@
+    --data ~/downloads/sample-inputs
+"""
+
+import argparse
+import logging
+import os
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, Optional
+
+from tqdm import auto
+
+from sparseml.base import Framework, detect_framework, execute_in_sparseml_framework
+from sparseml.benchmark.results import BatchBenchmarkResult, BenchmarkResults
+from sparseml.benchmark.serialization import BenchmarkConfig, BenchmarkInfo
+from sparseml.framework.info import FrameworkInferenceProviderInfo, FrameworkInfo
+from sparseml.utils import clean_path, create_parent_dirs
+from sparseml.utils.helpers import convert_to_bool
+
+
+__all__ = [
+    "BenchmarkRunner",
+    "save_benchmark_results",
+    "load_benchmark_info",
+    "load_and_run_benchmark",
+]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BenchmarkRunner(ABC):
+    """
+    Abstract class for handling running benchmarks with different frameworks.
+    """
+
+    def run(
+        self,
+        data: Any,
+        desc: str = "",
+        show_progress: bool = False,
+        *args,
+        **kwargs,
+    ) -> BenchmarkResults:
+        """
+        Runs a benchmark on the given data.
+
+        :param data: data to use for benchmarking
+        :param desc: str to display if show_progress is True
+        :param show_progress: whether to show progress
+        :param args: additional arguments to pass to the framework
+        :param kwargs: additional arguments to pass to the framework
+        :return: the results of the benchmark run
+        :rtype: BenchmarkResults
+        """
+        results = BenchmarkResults()
+        for batch_result in self.run_iter(
+            data, desc=desc, show_progress=show_progress, *args, **kwargs
+        ):
+            results.append(batch_result)
+        return results
+
+    def run_iter(
+        self,
+        data: Any,
+        desc: str = "",
+        show_progress: bool = False,
+        *args,
+        **kwargs,
+    ) -> Iterable[BatchBenchmarkResult]:
+        """
+        Iteratively runs a benchmark on the given data.
+
+        :param data_loader: data loader to use
+        :param desc: str to display if show_progress is True
+        :param show_progress: whether to show progress
+        :param args: additional arguments to pass to the framework
+        :param kwargs: additional arguments to pass to the framework
+        :return: an iterator of the benchmark results for each batch
+        :rtype: Iterable[BatchBenchmarkResult]
+        """
+        progress_steps = self.warmup_iterations + self.iterations
+        _LOGGER.debug("running {} items through model".format(progress_steps))
+        data_iter = (
+            enumerate(data)
+            if not show_progress
+            else auto.tqdm(enumerate(data), desc=desc, total=progress_steps)
+        )
+        for index, batch in data_iter:
+            if index < self.warmup_iterations:
+                continue
```

Comment:
we should still be running the batch for warmup here, correct?

Reply:
Ah sorry was supposed to run self.run_batch then continue, will fix